### PR TITLE
Student Profile backend

### PR DESCRIPTION
**Pull request by vipulrawat**
- Added 'request-cookies' package that extracts cookies from the request.
- Formatted timezone in Winston logs as Winston use ISO format.

### CODE REVIEW

File: packages/api/__tests__/index.js [hunk 1]
```diff
@@ -290,11 +291,13 @@ describe('/api/profile/createUser', () => {
         json: true,
         resolveWithFullResponse: true,
       }).then((res) => {
-        const cooky = res.headers['set-cookie'][0].substr(6, 200);
+        const rawcookies = res.headers['set-cookie'];
+        const cookieWithHeaders = new Cookie(rawcookies[0]);
```
Comment (ackinc) [inline packages/api/__tests__/index.js]: This is progress, but what if backend changes to set multiple cookies, and the token is no longer the first cookie in rawcookies? Better to not rely on non-explicit things like position.

File: packages/api/__tests__/index.js [hunk 2]
```diff
@@ -316,11 +319,13 @@ describe('/api/profile/createUser', () => {
...
```
Comment (ackinc) [inline packages/api/__tests__/index.js]: See above

File: packages/api/__tests__/index.js (hunk 1 above)
Comment (vipulrawat) [reply packages/api/__tests__/index.js]: ok. but will there be any other cookie apart from the login Token while logging in the user?

**Comment by ackinc** [comment]
It's really up to us/future devs. Cookies are just a general-purpose
client-side storage solution.

One example: we can store the start date of student's batch in a cookie on
profile creation. Then frontend can do stuff like "Congrats on passing your
second week!" without asking the backend for details of the user. Backend
routes that need this info would have it in the request itself, and
wouldn't have to query the DB.

Another example: Project Delta gets multiple-language support. The initial
language for a user is determined by the country + city selected during
profile creation, and stored client-side as a cookie (createUser route
returns the appropriate set-cookie header). Frontend changes this value
when user selects a different language, and backend looks at this value
when deciding what language to return text content (welcome message,
announcements etc.) in.

Obviously, these things can also be achieved without cookies. My opinion is
we should write code that supports any of the feasible choices. You can get
Arfat's and Bharani's and then make up your own mind about the best way to
handle this scenario.

File: packages/api/__tests__/index.js (hunk 1 above)
Comment (bharaninb) [reply packages/api/__tests__/index.js]: Simple solution:
`rawcookies.find(() => { // Your code })`

**Pull request closed without merging**